Homework 2: DB, Normalisation

whmi1 opened the pull request.

andrewD46 commented:
It would be better to add another entity – **groups**

andrewD46 commented:
You added a **UNIQUE** constraint to the foreign keys (FK) in the **schedule** and **academic_records** tables, which **completely breaks the logic**.

In the **schedule** table:

**subject_id UNIQUE** means that one subject can be in the schedule only once for the entire time. This is incorrect, because, for example, "Mathematics" can be in different groups at different times.

**teacher_id UNIQUE** means that one teacher can teach only one class for the entire time. This is also incorrect.

In the **academic_records** table:

**student_id UNIQUE** means that a student can have only one grade for the entire period of study.

**subject_id UNIQUE** means that only one grade can be given to all students for a subject.

**teacher_id UNIQUE** means that a teacher can give only one grade for his entire career.

andrewD46 commented:
The **schedule** table lists the **subject**, **teacher**, and **classroom**, but it is unclear who the class is for. **What group of students should attend?**

whmi1 commented:
Ok, I get it. Now I need to add these changes to the project on this branch, close this PR and make new or I can do it in this one?

andrewD46 commented:
> Ok, I get it. Now I need to add these changes to the project on this branch, close this PR and make new or I can do it in this one?

No, you just need to add those changes to that branch and that PR will be automatically updated

whmi1 commented:
Thanks a lot for answering the previous question!

Based on the results of the check in **db_document**, the following was done:
1. Tables **students_groups**, **faculties**, **departments**, and **academic_degrees**. These tables contain up-to-date information, the old tables have been updated, and foreign keys have been added to them to interact with the new tables. Field **group_id** also added into students and schedule tables.
2. Fixed a bug with fields **UNIQUE**. Now fields **subject_id**, **teacher_id** from the **schedule** table and fields **student_id**, **subject_id**, **teacher_id** from **academic_records** can use same data.
3. The **diagram** has been adjusted based on all of the above.

whmi1 commented:
Thank you for checking
